genet/schedule_elements.py

```diff
@@ -287,22 +372,74 @@ def __str__(self):
 
     def _build_graph(self, stops: List[Stop]):
         route_graph = nx.DiGraph(name='Route graph')
-        route_nodes = [(stop.id, stop.__dict__) for stop in stops]
+        try:
+            route_nodes = [(stop.id, stop.__dict__) for stop in stops]
+            stop_edges = [(from_stop.id, to_stop.id) for from_stop, to_stop in zip(stops[:-1], stops[1:])]
+        except AttributeError:
+            route_nodes = [(stop, {}) for stop in stops]
+            stop_edges = [(from_stop, to_stop) for from_stop, to_stop in zip(stops[:-1], stops[1:])]
```

Comment:
looks hacky, enables building a route graph only for the purpose of adding it to a schedule graph that already has stops. Don't worry, if they give a stop id that doesnt exist in the Schedule they will get a warning that there isnt any data for that stop when they try to add it (there is no other way to know before you try to add it, as you're only initiating this route her, it has no connection to the Schedule yet)